src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/AdjudicationMappingService.kt

```diff
@@ -11,21 +11,56 @@ import org.springframework.data.domain.Pageable
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.DuplicateMappingException
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.data.AdjudicationAllMappingDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.data.AdjudicationHearingMappingDto
 import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.data.AdjudicationMappingDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.data.AdjudicationPunishmentMappingDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.AdjudicationHearingMapping
 import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.AdjudicationMapping
 import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.AdjudicationMappingType
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.AdjudicationPunishmentMapping
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.repository.AdjudicationHearingMappingRepository
 import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.repository.AdjudicationMappingRepository
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.repository.AdjudicationPunishmentMappingRepository
 
 @Service
 @Transactional(readOnly = true)
 class AdjudicationMappingService(
   private val adjudicationMappingRepository: AdjudicationMappingRepository,
+  private val adjudicationHearingMappingRepository: AdjudicationHearingMappingRepository,
+  private val adjudicationPunishmentMappingRepository: AdjudicationPunishmentMappingRepository,
   private val telemetryClient: TelemetryClient,
 ) {
   private companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
+  @Transactional
+  suspend fun createMapping(createMappingRequest: AdjudicationAllMappingDto) {
+    createMapping(
+      createMappingRequest.adjudicationId.copy(
```

Comment:
Why the use of `copy` here?

Reply:
because we always want the mappingType and label to equal the parent mapping.
This is a special case where a migration can create *many* mapping records and they all need to either fail or succeed